src/sentry/web/frontend/oauth_token.py

```diff
@@ -116,17 +116,14 @@ def get_access_tokens(self, request: Request, application: ApiApplication) -> di
             return {"error": "invalid_grant", "reason": "invalid redirect URI"}
 
         token_data = {"token": ApiToken.from_grant(grant=grant)}
-        id_token = None
-        if grant.has_scope("openid"):
-            id_token = OpenIDToken(
+        if grant.has_scope("openid") and options.get("codecov.client-secret"):
+            open_id_token = OpenIDToken(
                 request.POST.get("client_id"),
                 grant.user_id,
-                # Encrypt with a random secret until we implement secure shared secrets in prod
-                secrets.token_urlsafe(),
+                options.get("codecov.client-secret"),
                 nonce=request.POST.get("nonce"),
             )
-            token_data["id_token"] = id_token.get_encrypted_id_token(grant=grant)
-
+            token_data["id_token"] = open_id_token.get_encrypted_id_token(grant=grant)
```

Comment:
Can we rename `get_encrypted_id_token` in this PR or should we wait for another?

The token isn't encrypted, it's just encoded + signed. The data within it is visible without the key. A more appropriate name would be `get_signed_id_token`.

Reply:
Very good point! Fixed